gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java

```diff
@@ -529,9 +529,18 @@ protected void cleanup() {
    */
   private DataWriter<Object> buildWriter()
       throws IOException {
+    String writerId = this.taskId;
+
+    // Add the task starting time if configured.
+    // This is used to reduce file name collisions which can happen due to the scheduling of a task on multiple workers.
+    // One case where this occurs is when a worker gets disconnected from a Helix cluster.
+    if (this.taskState.getPropAsBoolean(ConfigurationKeys.WRITER_ADD_TASK_TIMESTAMP, false)) {
+        writerId = this.taskId + "_" + this.taskState.getProp(ConfigurationKeys.TASK_START_TIME_MILLIS_KEY, "0");
```

Comment:
minor : not sure if all task execution will always come from `GobblinMultiTaskAttempt.java`, so can it use `Long.toString(System.currentTimeMillis())` instead of "0" as default? Also do we have to put `TASK_START_TIME_MILLIS_KEY` while beginning the task? can it not be just any current time when it comes here if its configured? ( start time would be helpful for other metrics though)